### DISCUSSION TITLE
V4 beta usage

### DISCUSSION THREAD

**Discussion by Eli-Nathan** [software-mansion/react-native-screens#2353] [Q&A] [answered]
The [V4 beta](https://github.com/software-mansion/react-native-screens/releases/tag/4.0.0-beta.0) is very exciting to see! 🎉 

Just wondering if there's an option to test it out?

Using it with the latest stable version of `@react-navigation/native-stack` (6.11.0) throws an exception when used with the RNScreens beta. The error is related to trying to transform "all" into an integer so it seems as though the previous `sheetAllowedDetents` contract is conflicting with the new one?

![Screenshot 2024-09-26 at 15 58 05](https://github.com/user-attachments/assets/ee03e930-6370-4f32-8f3a-71d6e846c6f4)

Is there a version of react-navigation that's compatible? And also do you have any more info on the minimum RN version required for the beta?

Thank you!

**Comment by Eli-Nathan** [comment]
I cannot get Android to run to test out the progress of the FormSheet support there either
The build fails due to the calls to `jsPointerDispatcher?.handleMotionEvent` in `java/com/swmansion/rnscreens/bottomsheet/BottomSheetDialogRootView.kt` having 3 arguments when it expects only 2? Would I need a specific RN version for this?

I could get it to build by removing the 3rd boolean params but then the app throws the error below.

I know the beta is very much a WIP so it could just be that it's not ready for testing yet but any pointers so I can give feedback would be awesome 😄 

![Screenshot 2024-09-26 at 16 00 33](https://github.com/user-attachments/assets/fa69107f-bc40-4127-bc35-0035842fee8d)

**Comment by shovel-kun** [comment, marked as answer]
@Eli-Nathan Use React Navigation v7.

**Reply by Eli-Nathan** [reply]
Thank you! React Navigation v7 relies on RN >= v0.72.0
